src/hdusd/__init__.py

```diff
@@ -21,15 +21,86 @@
     "location": "Info header, render engine menu",
     "description": "USD Hydra rendering plugin for Blender",
     "warning": "",
-    "tracker_url": "",
-    "doc_url": "",
-    "category": "Render"
+    "tracker_url": "https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon/issues",
+    "doc_url": "https://radeon-pro.github.io/RadeonProRenderDocs/en/usd_hydra/about.html",
+    "category": "Render",
+    "community": "https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon/discussions",
+    "downloads": "https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon/releases",
+    "main_web": "https://www.amd.com/en/technologies/radeon-prorender",
 }
 version_build = ""
 
 
+import tempfile
+from pathlib import Path
+from logging import getLevelName
+
 from . import config
-from .utils import logging
+from .utils import logging, temp_dir
+
+import bpy
+from bpy.types import AddonPreferences
+from bpy.props import StringProperty, BoolProperty, EnumProperty
+
+
+class HDUSD_ADDON_PT_preferences(AddonPreferences):
+    bl_idname = __name__
+
+    def update_temp_dir(self, value):
+        if not Path(self.tmp_dir).exists() or tempfile.gettempdir() == str(Path(self.tmp_dir)):
+            log.info(f"Current temp directory is {tempfile.gettempdir()}")
+            return
+
+        tempfile.tempdir = Path(self.tmp_dir)
+        bpy.context.preferences.addons[__name__].preferences['tmp_dir'] = str(temp_dir())
+        log.info(f"Current temp directory is changed to {bpy.context.preferences.addons[__name__].preferences.tmp_dir}")
+
+    def update_dev_tools(self, context):
+        config.show_dev_settings = self.dev_tools
+        log.info(f"Developer settings is {'enabled' if self.dev_tools else 'disabled'}")
+
+    def update_log_level(self, context):
+        logging.logger.setLevel(self.log_level)
+        log.critical(f"Log level is set to {self.log_level}")
```

Comment:
Why log.critical() is here? I think log.info() should be

Reply:
The purpose is to output update_log_level to a console regardless of the log level.